src/analysis/bounds.rs

```diff
@@ -122,8 +122,8 @@ impl Bounds {
                         let out_parameters = find_out_parameters(env, &function);
                         let parameters = format_out_parameters(&out_parameters);
                         let error_type = find_error_type(env, &function);
-                        type_string = format!("Fn(Result<{}, {}>) + Send + Sync + 'static", parameters, error_type);
-                        trampoline_type = Some(type_string.clone());
+                        type_string = format!("FnOnce(Result<{}, {}>) + Send + 'static", parameters, error_type);
+                        trampoline_type = Some((type_string.clone(), *self.unused.front().unwrap()));
```

Comment:
This part here is ugly. Any suggestions?

Reply:
Only see do `let bound_name = *self.unused.front().unwrap();` before
and rename to `trampoline_info`

Reply:
Done